src/rocm_docs/theme.py

```diff
@@ -24,19 +36,38 @@ def _update_repo_opts(srcdir: str, theme_opts: Dict[str, Any]) -> None:
         theme_opts.setdefault(key, val)
 
 
+def _set_theme_announcement(
+    announcement: str, theme_opts: Dict[str, Any]
+) -> None:
+    theme_opts.setdefault("announcement", announcement)
+
+
+def _update_banner(url: str, branch: str, theme_opts: Dict[str, Any]) -> None:
+    mapping_file_loc = "data/theme_announcement.yaml"
+    announcement_yaml = (
+        importlib_resources.files("rocm_docs") / mapping_file_loc
+    )
+
+    with open(announcement_yaml) as file:
+        banner_dict = yaml.safe_load(file)
+
+    if branch in banner_dict["old-branch"]:
+        _set_theme_announcement(banner_dict["old-str"], theme_opts)
+    elif branch in banner_dict["unreleased-branch"]:
+        _set_theme_announcement(banner_dict["unreleased-str"], theme_opts)
```

Comment:
I think this is not the place for determining if this is a release / release candidate / development_branch, projects.py is better suited for that IMO.

It could set a default key in the `html_theme_options` (to make it overridable) and the theme can react to that.

Reply:
[eae472c](https://github.com/RadeonOpenCompute/rocm-docs-core/pull/261/commits/eae472ce2e8de73321906e01fe3891828b12e1ca) moves this logic to projects.py

theme.py now just sets the announcement theme_opt